pkg/resources/hetzner/resources.go

```diff
@@ -269,16 +277,22 @@ func deleteLoadBalancer(cloud fi.Cloud, r *resources.Resource) error {
 }
 
 func deleteServer(cloud fi.Cloud, r *resources.Resource) error {
+	ctx := context.TODO()
+
 	klog.Infof("Deleting Server: %s(%s)", r.Name, r.ID)
 
 	c := cloud.(hetzner.HetznerCloud)
-	client := c.ServerClient()
+	serverClient := c.ServerClient()
 	server := r.Obj.(*hcloud.Server)
-	_, err := client.Delete(context.TODO(), server)
+	result, _, err := serverClient.DeleteWithResult(ctx, server)
 	if err != nil {
 		return fmt.Errorf("failed to delete server %s(%s): %w", r.Name, r.ID, err)
 	}
 
+	if _, err := hetzner.WaitForAction(ctx, c, result.Action); err != nil {
+		return fmt.Errorf("error waiting for server deletion: %w", err)
+	}
```

Comment:
Wouldn't using `actionClient.WatchProgress()` be simpler?
```suggestion
	_, errCh := actionClient.WatchProgress(ctx, result.Action)
	if err := <-errCh; err != nil {
		return err
	}
```

Reply:
That first return value is suspicious, because it seems like most of the messages to it will be dropped.  But as we're not using it, yes - good call...